### PR TITLE
Split the VM creation out of core

### PR DESCRIPTION
**Pull request by srsakhamuri**
This will facilitate adding of new VM provisioners for e.g. proxmox
in future

### CODE REVIEW

**Comment by rgreinho** [comment]
Although it looks nice, the build failed: https://travis-ci.org/rgreinho/molecule/builds/100663037

You need to run ``tox`` before creating a PR. I'll review it after that.

File: molecule/provisioners.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+#  THE SOFTWARE.
+
+import vagrant
```
Comment (abrown-sg) [inline molecule/provisioners.py]: You'll want to have a look at how we organize our imports. There's a convention we follow in all the files.

**Comment by rgreinho** [comment]
Great job! :+1: after fixing the imports and squashing the 2 commits.

**Comment by abrown-sg** [comment]
@srsakhamuri Really impressive work here! You'll want to rebase with a pretty large change to commands.py (moves each command to its own class) and have a look at the imports. Then we can have another look.

**Comment by srsakhamuri** [comment]
@abrown-sg Sure will do that, yeah I gone thru your changes should be easy to rebase. Thanks.

**Comment by srsakhamuri** [comment]
@abrown-sg Done rebasing and tested. Can you take a look?

**Comment by abrown-sg** [comment]
LGTM! I would like @retr0h to review as well.

**Comment by retr0h** [comment]
:+1:

**Pull request merged**